### DISCUSSION TITLE
Feature Request: Balance privacy for accounts with a phone number mapping

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/social-connect#208] [Feature requests (Needs)]
Surfacing this need from a recent conversation I had: 

> As a user that has mapped their phone number, I would like to be able to keep my current balance private from people that know my phone number.

**Comment by avaziri** [comment]
So many great features have been added to make Celo the best blockchain to transact on (not just store value!). I am talking about the stablecoins, the low fees, the fast block times, and the mobile-first features. Unfortunately, without balance privacy and private transactions, users would have to broadcast their wealth and their up-to-the-second transactions to the whole world. I often hear people compare Celo's mobile approach to Venmo. However, without privacy that comparison is not accurate; Celo is not currently usable for daily transactions.

My organization works with communities of unbanked people in extreme poverty. Unfortunately, in those settings fraud is common. Scammers would have a field day if everyone started broadcasting their wealth, and a live feed of all the businesses they had just visited and transacted with.

**Reply by avaziri** [reply]
Some common responses I get:
**Why don't people just off-ramp out of Celo and use cash for day-to-day transactions?**
That is likely going to be the solution for the foreseeable future. Unfortunately "just use cash" is antithetical to building strong Celo ecosystems, or people holding a balance in cryptocurrencies. Stopping here would basically admit that Celo won't try to be as usable as Venmo.

**Why don't you use Poof.cash?**
Poof.cash is a DApp that gives the ability to send and receive funds privately. This is the right *kind* of functionality but it exists in the wrong part of the tech stack. Poof is a DApp. Poof is not a wallet. Poof is privacy layered on top of everything else, rather than privacy at the foundation. Let me give some concrete examples of common tasks where Poof privacy breaks down:
1) I want to pay for my bill via a private transaction. If I send the payment via Poof, the money will come from a Poof smart contract and the merchant would be unable to tell which customer had settled their debt. I have no way to selectively disclose my identity to a trusted party for a single transaction (see this related feature celo-org/identity#4))
2) I want to privately receive money from a friend who knows my phone number. My phone number is associated with my public wallet address. Poof cant help me in this situation. I have to pick one or the other, mobile-ready or private.
3) I want to interact with some sort of DeFi DApp that requires me to hold funds and sign transactions with the same wallet (which is pretty much all DeFi DApps). Poof cant sign transactions. I will have to make a new one-use public wallet and send my poof balance to it. I will have to repeat this process for every DApp I use. If I mistakenly transact with one of those one-use wallets during my daily life I will de-anonymize my entire history of transactions.

Today in the Celo State of the Network Call I asked about privacy improvements. @marekolszewski (President of Valora) responded that (1) Valora has Poof available in the DApp directory (2) If Valora sees a lot of people using Poof they might consider rolling the functionality into Valora in a more seamless way. I would urge Marek to reconsider the wait-and-see approach. To use Poof presently for simple tasks is so onerous, and so incompatible with mobile-ready features that it will likely never catch on. We should be baking a privacy solution into the Celo blockchain. I cant understate how much more useful the entire ecosystem, and Valora in specific would be with seamless privacy.

**Reply by arthurgousset** [reply]
@avaziri Thank you for sharing the context above! I really appreciate it :) Upvotes and information like this really help us understand what to focus on.

**Reply by alecps** [reply]
@avaziri This is really helpful and I couldn't agree more. Transaction / balance privacy is a challenging problem (especially given the opposing concerns around AML and so on) but I agree that it's critical for Celo to deliver on its intended use case for all the same reasons you mentioned. We'll do our best to get more eyes on this

**Comment by arthurgousset** [comment]
Brief follow-up for visibility: @albertomartin shared a post about the technical priorities for cLabs: 
- [What’s next for Celo. A technical roadmap, with input from sessions at Celo Connect](https://forum.celo.org/t/whats-next-for-celo-a-technical-roadmap-with-input-from-sessions-at-celo-connect/3313)

It includes a sentence that might be relevant here:

> Private transactions and private balances are areas that we want to explore more, especially with compliance in mind. We would love to get a pulse from you all on interest here.

I linked the following two feature requests in the comment section (for visibility): 
- celo-org/celo-monorepo#9293
- celo-org/identity#4 

Feel free to comment too @avaziri!

**Comment by arthurgousset** [comment]
Some [additional Twitter sentiment](https://twitter.com/Pinotio_DeFi/status/1518360681573400576?s=20&t=I4OcQyU3E-1bqLThZIuP8w): 

<img width="400" alt="image" src="https://user-images.githubusercontent.com/46296830/165144892-fb3e46dd-9a98-4ecd-811d-db04c6d17059.png">

**Comment by arthurgousset** [comment]
When transferring this discussion topic:
- from: `celo-org/celo-monorepo/discussions`
- to: `celo-org/identity/discussions`

the upvotes were reset.

I'm including a screenshot of the upvotes for future reference: 

<img width="600" alt="image" src="https://user-images.githubusercontent.com/46296830/167673856-67ec4555-2817-4cae-960f-ef575276ceb0.png">
